src/main/java/codeu/controller/ServerStartupListener.java

```diff
@@ -31,6 +33,9 @@ public void contextInitialized(ServletContextEvent sce) {
       List<Message> messages = PersistentStorageAgent.getInstance().loadMessages();
       MessageStore.getInstance().setMessages(messages);
 
+      /* This lets the EventStore access the user, conversation, and message lists. */
+      EventStore.getInstance().setEvents(users, conversations, messages);
```

Comment:
Note, I think one limitation with this approach of EventStore encapsulating all the users/conversations/messages is that if a NEW user/conversation/message is created, the EventStore becomes outdated, it does NOT automatically get that NEW user/conversation/message.

And if you want to keep EventStore up-to-date, every time NEW user/conversation/message is created, you not only have to update UserStore/ConverstaionStore/MessageStore, you have to update the EventStore.

And if you want to persist all the events in the database, for example, every time a new User is created, that user is inserted into both 'user' table as well as 'event' table. So duplicated data not desirable.

Reply:
I was thinking about these things and I think updating the EventStore will be the best option. As far as duplicate data, I'm going to think about how to circumvent that, but in my opinion, the ease of dealing with just the event class and how much more clean it makes the code in the activity.jsp file outweighs the cons.